src/lib/components/image.svelte

```diff
@@ -9,13 +9,16 @@
 	}: {
 		src: string;
 		class?: string;
-		color_palette: number[][];
+		color_palette: string[];
 	} = $props();
 
 	let canvas_element = $state<HTMLCanvasElement>();
-
 	let image_loaded = $state(false);
 
+	function rgb_to_hex(r: number, g: number, b: number) {
+	    return "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+	};
```

Comment:
Do we need this function here?

I dont think adding it here would be good, make it a seperate function, with this signature

```ts

	function rgb_to_hex(r: number, g: number, b: number, alpha?:number)
```

Reply:
we dont need this anywhere else?
fine whatever :|